numba/targets/arraymath.py

```diff
@@ -2344,6 +2344,55 @@ def np_imag_impl(a):
 #----------------------------------------------------------------------------
 # Misc functions
 
+@overload(np.delete)
+def np_delete(arr, obj):
+    # Implementation based on numpy
+    # https://github.com/numpy/numpy/blob/af66e487a57bfd4850f4306e3b85d1dac3c70412/numpy/lib/function_base.py#L4065-L4267
+
+    if not isinstance(arr, (types.Array, types.Sequence)):
+        raise TypingError("arr must be either an Array or a Sequence")
+
+    def np_delete_impl(arr, obj):
+        arr = np.ravel(np.asarray(arr))
+        N = arr.size
+
+        keep = np.ones(N, dtype=np.bool_)
+        obj = np.array(obj)
+        keep[obj] = False
+        return arr[keep]
+
+    def np_delete_scalar_impl(arr, obj):
+        arr = np.ravel(np.asarray(arr))
+        N = arr.size
+        pos = obj
+
+        if (pos < -N or pos >= N):
+            raise IndexError('pos must be less than the len(arr)')
```

Comment:
```suggestion
            raise IndexError('obj must be less than the len(arr)')
```
Perhaps do ^ so that this references the variables used in the `np.delete` signature?